Packages/com.unity.inputsystem/InputSystem/Plugins/UI/ExtendedPointerEventData.cs

```diff
@@ -106,6 +117,63 @@ internal static int TouchIdFromPointerId(int pointerId)
         /// </summary>
         public Vector2 tilt { get; set; }
         */
+
+        internal void ReadDeviceState()
+        {
+            if (control.parent is Pen pen)
+            {
+                uiToolkitPointerId = GetPenPointerId(pen);
+                #if UNITY_2021_1_OR_NEWER
+                pressure = pen.pressure.EvaluateMagnitude();
+                azimuthAngle = (pen.tilt.ReadValue().x + 1) * Mathf.PI / 2;
+                altitudeAngle = (pen.tilt.ReadValue().y + 1) * Mathf.PI / 2;
+                twist = pen.twist.ReadValue() * Mathf.PI * 2;
+                #endif
+            }
+            else if (control.parent is TouchControl touchControl)
+            {
+                uiToolkitPointerId = GetTouchPointerId(touchControl);
+                #if UNITY_2021_1_OR_NEWER
+                pressure = touchControl.pressure.EvaluateMagnitude();
+                radius = touchControl.radius.ReadValue();
+                #endif
+            }
+            else if (control.parent is Touchscreen touchscreen)
+            {
+                uiToolkitPointerId = GetTouchPointerId(touchscreen.primaryTouch);
+                #if UNITY_2021_1_OR_NEWER
+                pressure = touchscreen.pressure.EvaluateMagnitude();
+                radius = touchscreen.radius.ReadValue();
+                #endif
+            }
+            else
+            {
+                uiToolkitPointerId = UIElements.PointerId.mousePointerId;
+            }
+        }
+
+        private static int GetPenPointerId(Pen pen)
+        {
+            var n = 0;
+            foreach (var otherDevice in InputSystem.devices)
+                if (otherDevice is Pen otherPen)
+                {
+                    if (pen == otherPen)
+                    {
+                        return UIElements.PointerId.penPointerIdBase +
+                            Mathf.Clamp(n, 0, UIElements.PointerId.penPointerCount - 1);
```

Comment:
Mathf.Min might be a tiny bit more intention revealing.

Reply:
Absolutely. Changed it, thanks!